addons/sale_project/models/sale_order_line.py

```diff
@@ -300,14 +300,11 @@ def _determine_project(so_line):
             """Determine the project for this sale order line.
             Rules are different based on the service_tracking:
 
-            - 'project_only': the project_id can only come from the sale order line itself
-            - 'task_in_project': the project_id comes from the sale order line only if no project_id was configured
-              on the parent sale order"""
+            - 'project_only' or 'task_in_project': the project_id can only come from the sale order line itself
+            """
 
-            if so_line.product_id.service_tracking == 'project_only':
+            if so_line.product_id.service_tracking in ['project_only', 'task_in_project']:
```

Comment:
I just realized this is all this method does, and it's not being overridden anywhere.
I think we can get rid of it and change the code below slightly:
```diff
diff --git a/addons/sale_project/models/sale_order_line.py b/addons/sale_project/models/sale_order_line.py
index 2f39a2193e84..ec68f1f35873 100644
--- a/addons/sale_project/models/sale_order_line.py
+++ b/addons/sale_project/models/sale_order_line.py
@@ -320,7 +320,9 @@ class SaleOrderLine(models.Model):
         # project_only, task_in_project: create a new project, based or not on a template (1 per SO). May be create a task too.
         # if 'task_in_project' and project_id configured on SO, use that one instead
         for so_line in so_line_new_project:
-            project = _determine_project(so_line)
+            project = False
+            if so_line.product_id.service_tracking in ['project_only', 'task_in_project']:
+                project = so_line.project_id
             if not project and _can_create_project(so_line):
                 project = so_line._timesheet_create_project()
                 if so_line.product_id.project_template_id:
```

Reply:
done